templates/infra.json

```diff
@@ -8,340 +8,426 @@
-                    "powerShellVersion": "~7"
-                }
-            }
-        },
-        {
-            "type": "Microsoft.Web/sites",
-            "apiVersion": "2020-06-01",
-            "name": "[variables('functionAppName-engine')]",
-            "location": "[parameters('location')]",
-            "kind": "functionapp",
-            "dependsOn": [
-                "[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]",
-                "[resourceId('Microsoft.Storage/storageAccounts', parameters('stgacctName'))]",
-                "[resourceId('Microsoft.AppConfiguration/configurationStores', variables('appConfigName'))]"
-            ],
-            "identity": {
-                "type": "SystemAssigned"
-            },
-            "properties": {
-                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]",
-                "clientAffinityEnabled": true,
-                "siteConfig": {
-                    "appSettings": [
+                            "type": "Microsoft.Web/serverfarms",
+                            "apiVersion": "2020-06-01",
+                            "name": "[variables('hostingPlanName')]",
+                            "location": "[resourceGroup().location]",
+                            "sku": {
+                                "name": "EP1",
+                                "tier": "ElasticPremium"
+                            },
+                            "properties": {
+                                "name": "[variables('hostingPlanName')]",
+                                "workerSize": 3,
+                                "workerSizeId": 3,
+                                "numberOfWorkers": 1,
+                                "maximumElasticWorkerCount": "20"
+                            }
+                        },
                         {
-                            "name": "AzureWebJobsStorage",
-                            "value": "[concat('DefaultEndpointsProtocol=https;AccountName=', parameters('stgacctName'), ';EndpointSuffix=', environment().suffixes.storage, ';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('stgacctName')), '2019-06-01').keys[0].value)]"
+                            "type": "Microsoft.Web/sites",
+                            "apiVersion": "2020-06-01",
+                            "name": "[variables('engineFuncName')]",
+                            "location": "[resourceGroup().location]",
+                            "kind": "functionapp",
+                            "dependsOn": [
+                                "[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]",
+                                "[resourceId('Microsoft.Storage/storageAccounts', variables('storAcctName'))]",
+                                "[resourceId('Microsoft.AppConfiguration/configurationStores', variables('appConfigName'))]"
```

Comment:
Since this depends on the app config, might improve readability to put the app config higher in the template

Reply:
Makes sense, can do (in a future update).